src/server.js

```diff
@@ -70,13 +71,50 @@ app.post('/api/npsResponses/create', promiseHandler(async req => {
   if (errors) {
     throw new Response(400, errors)
   }
+  console.log('Creating NPS response for: ' + email)
   await datastore.save({
     key: datastore.key('NpsResponse'),
     data: { ...data, email, timestamp: new Date() }
   })
   return new Response(201)
 }))
 
+/**
+ * @api {post} /api/npsResponses/firstTimestamp Record & return timestamp when user first accessed app
+ * @apiName firstVisitTimestamp
+ * @apiVersion 1.0.0
+ * @apiGroup Surveys
+ * @apiParam {String} currTimestamp Current timestamp; if first visit then will be recorded & returned as firstTimestamp
```

Comment:
My thought was that if there were a time delay between when Terra calls `firstTimestamps/record` and when Rex processes the call, that it could result in Rex recording a slightly later `firstTimestamp` than the current time when Terra called Rex. It's likely to be a negligible time difference.
Another consideration is that Terra uses the current time to compare the results of `/record` and `lastNpsResponse` so it is generated on the front end regardless. Again unlikely that any time difference would be significant enough to affect the behavior in a meaningful way.